### PR TITLE
Add CI - Run Tests (v4)

### LINKED ISSUE #321: Implement basic CI for running tests
A basic CI should be implemented that at a minimum runs tests, to prevent issues like #318. Later it can be extended for building multiple artifacts.

GitHub Actions is a great option, and is already enabled for this repo.

### PR DESCRIPTION
**Pull request by BrunoBernardino**
This makes it so that on every push (includes PRs), dependencies and tests run on GitHub Actions.

It also adds a small badge reporting the latest push/run on the `v4` branch.

Fixes #321

### CODE REVIEW

File: packages/core/test/account.ts [hunk 1]
```diff
@@ -35,6 +35,6 @@ suite("Account", () => {
         await account.unlock(password);
 
         assert.equal(bytesToHex(publicKey), bytesToHex(account.publicKey));
-        assert.equal(bytesToHex(privateKey), bytesToHex(account.privateKey));
+        assert.equal(bytesToHex(privateKey as Uint8Array), bytesToHex(account.privateKey as Uint8Array));
```
Comment (BrunoBernardino) [inline packages/core/test/account.ts:38]: This fixes one of the errors I was getting.

**Review by MaKleSoft — APPROVED** [review]

**Comment by BrunoBernardino** [comment]
The CI failures probably need #320 in order to not continue.

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+on: [push]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12.x
```
Comment (MaKleSoft) [inline .github/workflows/run-tests.yml:12]: Let's use node v16 since that is what production will run on

**Review by MaKleSoft — CHANGES_REQUESTED** [review]

File: .github/workflows/run-tests.yml (hunk 2 above)
Comment (BrunoBernardino) [reply .github/workflows/run-tests.yml:12]: If you tell me which specific version of `node` and `npm` you're on, I can work on #320 next with those set.

Comment (MaKleSoft) [reply .github/workflows/run-tests.yml:12]: Like I said lets use 16.x + the latest version of npm that goes with it 👍

**Pull request merged**